dashboard/app/models/script.rb

```diff
@@ -158,6 +158,13 @@ def hide_pilot_units
     }
 
   validates :published_state, acceptance: {accept: SharedCourseConstants::PUBLISHED_STATE.to_h.values.push(nil), message: 'must be nil, in_development, pilot, beta, preview or stable'}
+  validate :deeper_learning_courses_cannot_be_launched
+
+  def deeper_learning_courses_cannot_be_launched
+    if old_professional_learning_course? && (launched? || pilot?)
+      errors.add(:published_state, 'must be in_development or beta for a deeper learning course.')
```

Comment:
there are more states than this now, and I'm not sure what would happen if we ended up in one of the other ones: https://github.com/code-dot-org/code-dot-org/blob/5ee5b2ce43ce68ff9a52472ab281023dd69716be/lib/cdo/shared_constants/curriculum/shared_course_constants.rb#L10-L12

can you make this more specific to require in development or beta?

Reply:
We actually are using the deprecated state right now for a bunch of deeper learning courses. Example: https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/config/scripts_json/csp1-dlp-18.script_json#L15 . So I think what I need to update is the comments and error message. I will do that now

Reply:
Interesting. Are deprecated courses visible in the dropdowns?

Reply:
Only pilot or launched courses will show for uses in the dropdown https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/concerns/curriculum/assignable_course.rb. A levelbuilder or someone with an editor experiment would be able to see them in the dropdown. I was planning to filter them out in the future but wanted to make sure product had thought through everything first

Reply:
https://codedotorg.atlassian.net/browse/PLAT-1790